File: 백준/Bronze/2930. 가위 바위 보/가위 바위 보.py
import sys
input = sys.stdin.readline

wins = {'S': 'P', 'P': 'R', 'R': 'S'}

def score(sanggeun, friend):
    if sanggeun == friend:
        return 1
    elif wins[sanggeun] == friend:
        return 2
    else:
        return 0

R = int(input())
sanggeun = list(input().strip())
N = int(input())
friends = [list(input().strip()) for _ in range(N)]

real_score = 0
max_score = 0

for r in range(R):
    for n in range(N):
        real_score += score(sanggeun[r], friends[n][r])
    
    best = 0
    for choice in ['S', 'P', 'R']:
        round_score = sum(score(choice, friends[n][r]) for n in range(N))
        best = max(best, round_score)
    max_score += best

print(real_score)
print(max_score)

File: 백준/Bronze/2930. 가위 바위 보/가위 바위 보.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const wins = {'S': 'P', 'P': 'R', 'R': 'S'};

function score(sanggeun, friend) {
    if (sanggeun === friend) return 1;
    if (wins[sanggeun] === friend) return 2;
    return 0;
}

const R = Number(input.shift());
const sanggeun = input.shift().split('');
const N = Number(input.shift());
const friends = [];
for (let i = 0; i < N; i++) {
    friends.push(input.shift().split(''));
}

let realScore = 0;
let maxScore = 0;

for (let r = 0; r < R; r++) {
    for (let n = 0; n < N; n++) {
        realScore += score(sanggeun[r], friends[n][r]);
    }
    
    let best = 0;
    for (const choice of ['S', 'P', 'R']) {
        const roundScore = friends.reduce((acc, f) => acc + score(choice, f[r]), 0);
        best = Math.max(best, roundScore);
    }
    maxScore += best;
}

console.log(realScore);
console.log(maxScore);